CMakeLists.txt

```diff
@@ -37,3 +47,23 @@ add_subdirectory(src/go)
 add_subdirectory(src/c)
 
 install(DIRECTORY "pixmaps" DESTINATION "${PURPLE_DATA_DIR}" FILES_MATCHING PATTERN "*.png")
+
+# package settings
+set(CPACK_PACKAGE_DESCRIPTION_SUMMARY "Whatsapp protocol plug-in for libpurple with pidgin")
+set(CPACK_PACKAGE_VENDOR "Bryant Eadon")
+set(CPACK_PACKAGE_DESCRIPTION "whatsmeow is a go-based whatsapp protocol plug-in for libpurple based on the whatsapp business apis.")
+set(CPACK_PACKAGE_CONTACT "bryant@balancebitsconsulting.com")
+set(CPACK_PACKAGE_VERSION_MAJOR "${VERSION_MAJOR}")
+set(CPACK_PACKAGE_VERSION_MINOR "${VERSION_MINOR}")
+set(CPACK_PACKAGE_VERSION_PATCH "${VERSION_PATCH}")
+set(CPACK_DEBIAN_PACKAGE_SHLIBDEPS ON)
+set(CPACK_DEBIAN_PACKAGE_ARCHITECTURE "${ARCHITECTURE}")
+set(CPACK_SOURCE_PACKAGE_FILE_NAME "${CMAKE_PROJECT_NAME}_${VERSION}_${CPACK_DEBIAN_PACKAGE_ARCHITECTURE}")
+set(CPACK_PACKAGE_FILE_NAME "${CMAKE_PROJECT_NAME}_${VERSION}_${CPACK_DEBIAN_PACKAGE_ARCHITECTURE}")

```

Comment:
This is pretty lengthy and will get longer as we consider more use-cases. I would like to have it in a separate file. I propose an outline like [this](https://github.com/hoehermann/purple-gowhatsapp/tree/cpack). Does that look okay to you?

Reply:
I can;t open the example you have unfortunately -- I have seen no example of people doing this in separate files.

The packaging files are very tightly coupled with build instructions, versions and more refer to how the binaries are built.  If it feels like clutter, perhaps there is a way to roll it up in an editor like VSCode ? or vim folds ?

All guides I have read so far are recommending including CPACK at the end of the CMakeLists.txt.

Reply:
I am one of those weird guys who code with a dumb text editor (syntax highlighting only). I moved the cpack commands into a separate file which is included by the main `CMakeLists.txt`. Seems to work so far.

Reply:
yes, that's fine.  

As for editors and style choices, I find a longer make file with everything in it is a reminder of the variables that you have at your fingertips.  Either other targets already built, or style choices you should be matching when writing new/different targets.

Anyway, this include method should work fine.